typescript/lib/parsers/openai.ts

```diff
@@ -700,17 +720,32 @@ export class OpenAIChatModelParser extends ParameterizedModelParser<Chat.ChatCom
     const output = aiConfig.getLatestOutput(prompt);
     if (output != null) {
       if (output.output_type === "execute_result") {
-        // If the prompt has output saved, add it to the messages array
         if (output.metadata?.rawResponse?.role === "assistant") {
-          if (typeof output.data === "string") {
-            messages.push({
-              content: output.data,
-              // TODO (rossdanlm): Support function_call and don't rely on rawResponse
-              role: output.metadata?.rawResponse?.role,
-              function_call: output.metadata?.rawResponse?.function_call,
-              name: output.metadata?.rawResponse?.name,
-            });
+          output as ExecuteResult;
+          let content: string | null = null;
+          let function_call: FunctionData | undefined = undefined;
```

Comment:
```suggestion
          let function_call: FunctionData | undefined;
```
nit, the value will default to `undefined` without needing to set it